### ISSUE TITLE
GIF conversion broken due to simple FreeImage misconfiguration

### ISSUE THREAD

**Issue by chronometric** [Nandaka/PixivUtil2#157]
The ugoira to GIF conversion implemented in issue #135 is broken in the 10-12 download, but only trivially. When it encounters a ugoira file, this is what happens:
```
Creating ugoira archive => D:\Artist - 123456789.ugoira
Deleting zip file => D:\Artist - 123456789.zip
processing ugoira...
Imageio: 'freeimage-3.15.4-win32.dll' was not found on your computer; downloading it now.
Error while fetching file: (0, 'connection closed unexpectedly').
Error while fetching file: (0, 'connection closed unexpectedly').
Error while fetching file: (0, 'connection closed unexpectedly').
Error while fetching file: (0, 'connection closed unexpectedly').
```
[**Previous 5 lines repeat 36 times**]
```
Traceback (most recent call last):
  File "PixivUtil2.py", line 692, in process_image
  File "PixivHelper.pyc", line 655, in ugoira2gif
  File "imageio\core\functions.pyc", line 181, in imread
  File "imageio\core\functions.pyc", line 108, in get_reader
ValueError: Could not find a format to read the specified file in mode 'i'
Error at process_image(): (<type 'exceptions.ValueError'>, ValueError("Could not
 find a format to read the specified file in mode 'i'",), <traceback object at 0
x0404B260>)
Dumping html to: Error medium page for image 123456789.html
Stuff happened, trying again after 2 second ( 1 )
Traceback (most recent call last):
  File "PixivUtil2.py", line 412, in process_member
  File "PixivUtil2.py", line 692, in process_image
  File "PixivHelper.pyc", line 655, in ugoira2gif
  File "imageio\core\functions.pyc", line 181, in imread
  File "imageio\core\functions.pyc", line 108, in get_reader
ValueError: Could not find a format to read the specified file in mode 'i'
Processing Image Id: 123456789
Title: ????????????
Tags : ??? ???? ???? ????? Fate/GrandOrder, Live2D, ?????????????????
Date : 2016-10-05 17:58:00
Mode : ugoira_view
Image URL : http://i1.pixiv.net/img-zip-ugoira/img/2016/10/05/17/58/37/123456789_
ugoira1920x1080.zip
Filename  : D:\Artist - 123456789.zip
Checking local filename... Getting remote filesize...
Using Referer: http://www.pixiv.net/member_illust.php?mode=medium&illust_id=123456789
        File exist! (Identical Size)
		
#2
Processing Image Id: <next>
```

The FreeImage library is mentioned without explanation in the readme file. It turns out the required DLL is available [here] (https://github.com/imageio/imageio-binaries/blob/master/FreeImage-3.15.4-win32.dll). The function works again after placing this file in the Windows\System32 directory.

This was a bit confusing because the current version is 3.17 and is on the project's [main page](http://freeimage.sourceforge.net/download.html), but causes PixivUtil to ignore Ugoira files.

Could this file be distributed with the program libraries? Or at least with the program, with a better explanation in the readme? :3

**Comment by Nandaka** [comment]
>> The function works again after placing this file in the Windows\System32 directory

What about placing it on the app folder?I'm not comfortable placing dll on system32.
Also related for non-windows system.

>> current version is 3.17 

Is it not working with 3.17? If yes, then I can add freeimage-3.15.4-win32.dll to the repo.

**Comment by chronometric** [comment]
Indeed, first I tried 3.17 in the pixivutil program directory, then in the system32/64 folders.

Neither of them worked in the app folder. As I deleted all the animated files in a very long directory I was downloading, this case would skip over the ugoira files entirely and then move on. Only when I placed 3.15.4 in the system folders did it correctly re-download the animations I have previously deleted, and convert them directly to JPEG.

So speaking for windows, adding 3.15.4 to the repo would seem to be exactly what it craves. Mac and linux might require their own testing of this situation, because the only versions that were archived in the imageio-binaries project were 3.10.x, and I have no clue what in the coding made windows demand the older version it did (and since it tried to download files that are museum archived, it might indicate there's a block of old code somewhere here). 

If OSX and linux don't happen to call for the 3.10.x version archived on github, it might be simplest (or at least, the only) solution to just bring everything up to speed at 3.70. Though four years have passed, the [http://freeimage.sourceforge.net/changeslog.html](changelog) seems to be a mixture of high-end image processing/format usage and legacy capabilities. 

Though <[Mihail Naydenov] fixed GIF plugin LZW decoder failing on some image> is one of the first things they fixed after 3.15.4 ^^;;;; Or does PixivUtil use the library for anything besides the animated GIF conversion?
